operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResource.java

```diff
@@ -12,4 +13,46 @@ default EventSource initEventSource(EventSourceContext<P> context) {
   default Class<R> resourceType() {
     return (Class<R>) Utils.getFirstTypeArgumentFromInterface(getClass());
   }
+
+  /**
+   * todo: return Optional<R></R> instead?
```

Comment:
Yes, I think should be optional, we should never use `null` if it's avoidable

Reply:
Done.